### PR TITLE
More robust dev scripts and removed util directory

### PR DESCRIPTION
**Pull request by cmnrd**
This PR removes the util directory. By removing the utli directory, I hope to reduce the incentive on dumping unfurnished code and experimental tools in the main LF repo.

The only content of util at the moment are the launch helper scripts. This PR removes the launch scripts and instead implements the logic directly in the `lf*-dev` scripts. This adds a bit of code duplication (the readlink logic), but also simplifies other aspects of the script. Overall, the new solution should be more robust.

This PR also adds back the capability to run the `lf*-dev` scripts from anywhere in the filesystem. This was removed by  #1988.

Finally, this PR also fixes bugs in the way we handled scripts before.

### CODE REVIEW

**Comment by cmnrd** [comment]
It also turns out that there was a major bug in the readlink logic. I honestly don't know why this passed tests before.

File: .github/scripts/test-lfc.sh [hunk 1]
```diff
@@ -18,3 +18,8 @@ bin/lfc-dev test/C/src/Minimal.lf
 
 # Ensure that lfc can be invoked via symbolic links.
 test_with_links "lfc-dev"
+
+# Ensure that lfc can be invoked from outside the root directory.
+cd bin
+./lfc-dev --help
```
Comment (lhstrh) [inline .github/scripts/test-lfc.sh:24]: Why not skip the `cd`s and use a relative path, like this?
```suggestion
./bin/lfc-dev --help
```

File: .github/scripts/test-lfc.sh [hunk 2]
```diff
@@ -18,3 +18,8 @@ bin/lfc-dev test/C/src/Minimal.lf
 
 # Ensure that lfc can be invoked via symbolic links.
 test_with_links "lfc-dev"
+
+# Ensure that lfc can be invoked from outside the root directory.
```
Comment (lhstrh) [inline .github/scripts/test-lfc.sh:22]: Should we also add a test to execute outside of the dev tree? (`./bin` is still inside)

**Review by lhstrh — APPROVED** [review]
Modulo a minor comment about the tests, this looks like the right solution to me.

**Comment by lhstrh** [comment]
> It also turns out that there was a major bug in the readlink logic. I honestly don't know why this passed tests before.
> 
> Edit: Actually, I do know: The tests passed before because both the launch scripts and the link structure crated in cli-test is 2 levels deep. It resolved the wrong path, but the end result was correct coincidentally...

:open_mouth:

**Pull request merged**